src/files.rs

```diff
@@ -1,29 +1,55 @@
 use std::process::Command;
 use std::{fs, str::FromStr};
 
-#[derive(PartialEq)]
+#[derive(PartialEq, Clone, Copy)]
 pub enum FileType {
     Xml,
     Ifcfg,
+    Ifroute,
+    Route,
+}
+
+impl FileType {
+    pub fn file_types_are_compatible(type1: Option<FileType>, type2: Option<FileType>) -> bool {
+        let type1 = match type1 {
+            Some(type1) => type1,
+            None => return false,
+        };
+
+        let type2 = match type2 {
+            Some(type2) => type2,
+            None => return false,
+        };
+
+        if type1 == FileType::Xml && type2 == FileType::Xml
+            || type1 != FileType::Xml && type2 != FileType::Xml
+        {
+            return true;
+        }
+        false
+    }
```

Comment:
I don't see a reason to add more file types for these, since there is no special handling required here, these all should just be handled the same way, so I would probably just rename the ifcfg to something more generic like e.g. `Sysconfig` or smth. and that removes the need for this `file_types_are_compatible` function, which I find pretty ugly.